lib/gollum-lib/filter/toc.rb

```diff
@@ -24,15 +25,38 @@ def process(data)
         add_anchor_to_header header, anchor_name
         add_entry_to_toc     header, anchor_name
       end
+      if not @toc_doc.nil?
+        toc_str = @toc_doc.to_xml(@markup.to_xml_opts)
+      end
 
-      @toc  = @toc.to_xml(@markup.to_xml_opts) if @toc != nil
       data  = @doc.to_xml(@markup.to_xml_opts)
-      
     end
- 
-    @markup.toc = @toc
-    data.gsub("[[_TOC_]]") do
-      @toc.nil? ? '' : @toc
+
+    @markup.toc = toc_str
+
+    data.gsub(/\[\[_TOC_(.*?)\]\]/) do
```

Comment:
Could we replace this `gsub` call with `gsub!` without issues? If so, then we should do it.